sdk/storage/Azure.Storage.Blobs/src/BlobContainerClient.cs

```diff
@@ -1216,6 +1216,11 @@ private void SetNameFieldsIfNull()
                         response.ToBlobContainerInfo(),
                         response.GetRawResponse());
                 }
+                catch (RequestFailedException storageRequestFailedException)
+                when (storageRequestFailedException.ErrorCode == BlobErrorCode.ContainerAlreadyExists && !failScopeOnAlreadyExists)
```

Comment:
I'm confused how this resolves the issue. You moved this catch-when statement to the out of the `CreateIfNotExistsInternal` and into `CreateInternal`, and then added a boolean to check if `CreateIfNotExistsInternal` was the one calling this statement in order to catch the exception.

I don't see any behavior change being made here.

You also don't have any tests added as well, so I can see the scenario being reproduced here.

Reply:
The difference is that before this change, this method would catch the exception in the catch handler below, which would log an exception and do a `scope.fail()`, which was the "bug". 

In the new change, when it comes from CreateIfNotExistsAsync, when it throws an 409 exception from the API, it will no longer say in the logs that the Create dependency "failed", it will still show that the dependency returned a 409, but it won't have the red highlighting, and Azure Devops won't see it as a failed dependency.

Reply:
I think I would defer to Christopher's approach he replied below in the PR.